packages/components/src/date-time/time.js

```diff
@@ -89,7 +89,7 @@ class TimePicker extends Component {
 		const month = selected.format( 'MM' );
 		const year = selected.format( 'YYYY' );
 		const minutes = selected.format( 'mm' );
-		const am = selected.format( 'A' );
+		const am = selected.format( 'A' ) === __( 'AM' ) ? 'AM' : 'PM';
```

Comment:
I think it's better to compare it to the numbers.
If the translation in moment.js is different from the translation in WordPress, a problem occurs.

Reply:
It is better to compare it with `selected.format("H")` .